util/helm/client.go

```diff
@@ -381,3 +383,151 @@ func getIndexURL(rawURL string) (string, error) {
 	repoURL.RawPath = path.Join(repoURL.RawPath, indexFile)
 	return repoURL.String(), nil
 }
+
+func getTagsListURL(rawURL string, chart string) (string, error) {
+	tagsList := path.Join("v2", chart, "tags/list")
+	repoURL, err := url.Parse(rawURL)
+	if err != nil {
+		return "", fmt.Errorf("unable to parse repo url: %v", err)
+	}
+	repoURL.Scheme = "https"
+	repoURL.Path = path.Join(repoURL.Path, tagsList)
+	repoURL.RawPath = path.Join(repoURL.RawPath, tagsList)
```

Comment:
```suggestion
func getTagsListURL(rawURL string, chart string) (string, error) {
	repoURL, err := url.Parse(rawURL)
	if err != nil {
		return "", fmt.Errorf("unable to parse repo url: %v", err)
	}
	repoURL.Scheme = "https"
	tagsList, err := url.JoinPath("v2", url.PathEscape(chart), "tags/list")
	if err != nil {
		return "", fmt.Errorf("unable to join chart name with tags/list path: %w", err)
	}
	path, err := url.JoinPath(repoURL.Path, tagsList)
	if err != nil {
		return "", fmt.Errorf("unable to join repo base path with tags/list path: %w", err)
	}
	rawPath, err := url.JoinPath(repoURL.RawPath, tagsList)
	if err != nil {
		return "", fmt.Errorf("unable to join raw repo base path with tags/list path: %w", err)
	}
	repoURL.Path = path.Join(repoURL.Path, tagsList)
	repoURL.RawPath = path.Join(repoURL.RawPath, tagsList)
```

I believe we should be using the `url` package instead of `path` for this.